compose/snippets/src/main/java/com/example/compose/snippets/text/TextSnippets.kt

```diff
@@ -862,31 +867,43 @@ val Purple = Color(0xFF800080)
 // [START android_compose_text_showhidepassword]
 @Composable
 fun PasswordTextField() {
-    var password by rememberSaveable { mutableStateOf("") }
+    val state = remember { TextFieldState() }
     var showPassword by remember { mutableStateOf(false) }
-    val passwordVisualTransformation = remember { PasswordVisualTransformation() }
-
-    TextField(
-        value = password,
-        onValueChange = { password = it },
-        label = { Text("Enter password") },
-        visualTransformation = if (showPassword) {
-            VisualTransformation.None
+    BasicSecureTextField(
+        state = state,
+        textObfuscationMode =
+        if (showPassword) {
+            TextObfuscationMode.Visible
         } else {
-            passwordVisualTransformation
+            TextObfuscationMode.RevealLastTyped
         },
-        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
-        modifier = Modifier.fillMaxWidth(),
-        trailingIcon = {
-            Icon(
-                if (showPassword) {
-                    Icons.Filled.Visibility
-                } else {
-                    Icons.Filled.VisibilityOff
-                },
-                contentDescription = "Toggle password visibility",
-                modifier = Modifier.clickable { showPassword = !showPassword }
-            )
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(6.dp)
+            .border(1.dp, Color.LightGray, RoundedCornerShape(6.dp))
+            .padding(6.dp),
```

Comment:
Looks like the padding modifier is set twice.

Reply:
This is intentional to add the padding after and before the border :)